docs/src/dev/profiling.md

```diff
@@ -60,47 +60,80 @@ Fields and their meaning are as follows:
 
 ## Perf
 
-Note that `yk-config --cflags` includes `-Wl,--no-rosegment`, which [gives
-better profiling
-information](https://github.com/flamegraph-rs/flamegraph#cargo-flamegraph) for
-binaries linked with `lld` (as all yk C interpreters are). 
+On Linux, `perf` can be used to profile yk. You must first need to record an
+execution of an interpreter and then separately view the profiling data that
+was generated.
 
-One way to view profile data is with perf.
+
+### Recording a Profile
+
+To record a profile we first recommend compiling yk with debugging info
+embedded. cargo's `debug` profile does this automatically, but because no code
+optimisation is performed, the profiles are unrepresentative. We recommend
+using yk's provided `profile-with-debug` profile, which turns on
+`--release`-style code optimisation *and* embeds debugging information:
+
+```
+$ cargo build --profile=release-with-debug
+```
+
+Ensure that the interpreter you are profiling links to the appropriate version
+of yk and then call `perf record`:
 
 ```
 $ perf record --call-graph dwarf -g ./interpreter ...args...
-$ perf report -G --no-inline
 ```
 
-`--call-graph=dwarf` makes perf use DWARF debugging information, which gives
-better quality profiling data than the default, but requires that you compile
-with debugging info.
+This uses `--call-graph dwarf` to force perf use DWARF debugging information:
+this will only be useful if you have compiled yk with embedded debugging
+information, as recommended above.
+
 
-We recommend passing `--no-inline` since, for all but the smallest runs,
```

Comment:
It's fast now?

Reply:
Not on b16... Note that this part has just moved elsewhere, so the warning is still present.

Reply:
I see it moved.